experimental/web_dashboard/lib/src/pages/sign_in.dart

```diff
@@ -20,24 +20,48 @@ class SignInPage extends StatefulWidget {
 }
 
 class _SignInPageState extends State<SignInPage> {
+  Future<bool> _checkSignInFuture;
+
   @override
   void initState() {
     super.initState();
+    _checkSignInFuture = _checkIfSignedIn();
+  }
+
+// Check if the user is signed in. If the user is already signed in (for
+// example, if they signed in and refreshed the page), invoke the `onSuccess`
+// callback right away.
+  Future<bool> _checkIfSignedIn() async {
```

Comment:
Because this is a sample, you wouldn't be incorrect to remove this section for simplicity's sake and instead expect users to hit the button each time they want to login.

Totally optional, but I figured I'd put it out there.

Reply:
This is certainly reaching the point of diminishing returns. It might be worth taking out and capturing this pattern as an article or something.